cmd/otelcol/main.go

```diff
@@ -41,9 +47,17 @@ func main() {
 		GitHash:  version.GitHash,
 	}
 
-	svc, err := service.New(service.Parameters{ApplicationStartInfo: info, Factories: factories})
-	handleErr("Failed to construct the application", err)
+	return service.Parameters{Factories: factories, ApplicationStartInfo: info}, nil
+}
 
-	err = svc.Start()
-	handleErr("Application run finished with error", err)
+func runInteractive(params service.Parameters) {
+	app, err := service.New(params)
+	if err != nil {
+		log.Fatalf("failed to construct the application: %v", err)
```

Comment:
We shouldn't use `log.Fatal` outside of the main function. See cleanup issue #619 where I have previously documented why. Please propagate errors back to the main function instead.

I have also documented this requirement in https://github.com/open-telemetry/opentelemetry-collector/blob/master/CONTRIBUTING.md#propagate-errors-to-the-caller